appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -2,448 +2,421 @@ package nu.yona.server
 
 		when:
-			def response = appService.updateResource(richardQuinURL, richardQuinCreationJSON, ["Yona-Password": "New password"], ["tempPassword": "hack"])
+			appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, updatedBob)
 
 		then:
-			response.status == 400
+			def getDirectMessagesResponse = appService.getDirectMessages(updatedBob)
+			getDirectMessagesResponse.status == 200
+
+		cleanup:
+			appService.deleteUser(richard)
+			appService.deleteUser(updatedBob)
 	}
 
-	def 'Bob Dunn downloads the app and opens the link sent in the email with the app; app retrieves data to prefill'(){
+	def 'Bob accepts Richard\'s buddy request'()
+	{
 		given:
+			def richard = addRichard()
+			def mobileNumberBob = "+$timestamp"
+			def inviteURL = sendBuddyRequestForBob(richard, mobileNumberBob).responseData.userCreatedInviteURL
+			def bob = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBoddyRequest, inviteURL, true, null)
+			def newNickname = "Bobby"
+			def newPassword = "B o b"
+			def updatedBobJson = bob.convertToJSON()
+			updatedBobJson.nickname = newNickname
+			updatedBobJson.devices = [ "iPhone 6" ]
+			updatedBobJson.goals = [ "gambling" ]
+			def updatedBob = appService.updateUser(appService.&assertUserUpdateResponseDetails, new User(updatedBobJson, newPassword), inviteURL);
+			appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, updatedBob)
+			def acceptURL = appService.fetchBuddyConnectRequestMessage(updatedBob).acceptURL
 
 		when:
-			def response = appService.getUser(bobDunnInviteURL, true, null)
+			def response = appService.postMessageActionWithPassword(acceptURL, ["message" : "Yes, great idea!"], newPassword)
 
 		then:
 			response.status == 200
-			response.responseData.firstName == "Bob ${timestamp}"
-			response.responseData.lastName == "Dun ${timestamp}"
-			response.responseData.mobileNumber == "+${timestamp}12"
-			response.responseData?.mobileNumberConfirmed == false
-			response.responseData.vpnProfile.vpnLoginID ==~ /(?i)^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/
-			response.responseData.vpnProfile.vpnPassword.length() == 32
-			response.responseData.vpnProfile.openVPNProfile.length() > 10
+			def bobWithBuddy = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBoddyRequest, updatedBob.url, true, updatedBob.password)
+			bobWithBuddy.buddies != null
+			bobWithBuddy.buddies.size() == 1
+			bobWithBuddy.buddies[0].user.firstName == "Richard"
+			bobWithBuddy.buddies[0].sendingStatus == "ACCEPTED"
+			bobWithBuddy.buddies[0].receivingStatus == "ACCEPTED"
 
+		cleanup:
+			appService.deleteUser(richard)
+			appService.deleteUser(updatedBob)
 	}
 
-	def 'Bob Dunn adjusts data and submits; app saves with new password'(){
+	def 'Richard processes Bob\'s buddy acceptance'()
```

Comment:
That would imply if-statements in the previous test: if the first step is successful, do the second one and verify the results. The good thing of the current test structure is that there are nearly no conditions in the tests. That increases the readability a lot.